### PR TITLE
Updating Display::DString

### PR DESCRIPTION
**Pull request by ineiti**
* adds `cursorHome`, `cursorVisible`, and `cursorDelete` to `DString`
* adds `cursor{Back,Forward,Up,Down,Position,{Next,Previous}Line,HorizontalAbsolute}`
* overloads `append` to return a reference to `DString`
* refactoring CSI string
* adds `<<(char)`
* move `CHANGELOG.md` to `README.md`

### CODE REVIEW

File: src/display.cpp [hunk 1]
```diff
@@ -97,14 +103,14 @@ namespace Display {
...
     }
 
     DString &DString::saveCursorPosition() {
         // While the first is the official ANSI code for saving the cursor position,
         // the latter works in most unix terminals, but not MacOS.
 #ifdef __WIN32__
-        *this += "\x1b[s";
+        *this += CSI + "s";
 #elif __linux__
         *this += "\x1b"
```
Comment (A2va) [inline src/display.cpp:107]: Use CSI

**Review by A2va — APPROVED** [review]

**Comment by A2va** [comment]
It seems that ANSI escape code has can also move the cursor relatively (up, down, left, right) or specific position (line and column).
Maybe it's the right time to add that as well.

File: src/display.cpp (hunk 1 above)
Comment (ineiti) [reply src/display.cpp:107]: Interesting - I think I pushed an old version, but cannot find the one where I forgot the CSI...

**Comment by ineiti** [comment]
> It seems that ANSI escape code has can also move the cursor relatively (up, down, left, right) or specific position (line and column). Maybe it's the right time to add that as well.

OK - added all cursor movements `A` to `H`, according to https://en.wikipedia.org/wiki/ANSI_escape_code

Can you check it's OK and then approve again, please?

**Review by A2va — APPROVED** [review]

**Pull request merged**